jf_agent/main.py

```diff
@@ -366,37 +370,53 @@ def download_data(config, creds, endpoint_jira_info, endpoint_git_instances_info
             print_all_jira_fields(config, jira_connection)
         download_data_status.append(load_and_dump_jira(config, endpoint_jira_info, jira_connection))
 
-    is_multi_git_config = len(config.git_configs) > 1
-    for git_config in config.git_configs:
-        agent_logging.log_and_print(
-            logger,
-            logging.INFO,
-            f'Obtained {git_config.git_provider} configuration, attempting download...',
-        )
-        if is_multi_git_config:
-            instance_slug = git_config.git_instance_slug
-            instance_info = endpoint_git_instances_info.get(instance_slug)
-            instance_creds = creds.git_instance_to_creds.get(instance_slug)
-        else:
-            # support legacy single-git support, which assumes only one available git instance
-            instance_info = list(endpoint_git_instances_info.values())[0]
-            instance_creds = list(creds.git_instance_to_creds.values())[0]
+    
+    # git downloading is parallelized by the number of configurations. 
+    futures = []
+    with ThreadPoolExecutor(max_workers=8) as executor:
+        for git_config in config.git_configs:
+            agent_logging.log_and_print(
+                logger,
+                logging.INFO,
+                f'Obtained {git_config.git_provider}:{git_config.git_instance_slug} configuration, attempting download {"in parallel" if len(config.git_configs) > 1 else ""}...',
+            )
+            futures.append(
+                executor.submit(
+                    _download_git_data,
+                    git_config,
+                    config,
+                    creds,
+                    endpoint_git_instances_info,
+                    len(config.git_configs) > 1,
+                )
+            )
 
-        git_connection = get_git_client(
-            git_config, instance_creds, skip_ssl_verification=config.skip_ssl_verification
-        )
+    return download_data_status + [f.result() for f in futures]
 
-        download_data_status.append(
-            load_and_dump_git(
-                config=git_config,
-                endpoint_git_instance_info=instance_info,
-                outdir=config.outdir,
-                compress_output_files=config.compress_output_files,
-                git_connection=git_connection,
-            )
-        )
 
-    return download_data_status
+def _download_git_data(
+    git_config, config, creds, endpoint_git_instances_info, is_multi_git_config
+) -> dict:
+
+    if is_multi_git_config:
+        instance_slug = git_config.git_instance_slug
+        instance_info = endpoint_git_instances_info.get(instance_slug)
+        instance_creds = creds.git_instance_to_creds.get(instance_slug)
+    else:
+        # support legacy single-git support, which assumes only one available git instance
+        instance_info = list(endpoint_git_instances_info.values())[0]
+        instance_creds = list(creds.git_instance_to_creds.values())[0]
```

Comment:
Just want to make sure this legacy path has been tested too (refactor looks right to my eye though)

Reply:
👍  I have tested configs of the form: 
```
git:
  provider: github
  url: https://api.github.com
  ....
```
and those of the form 
```
git:
  - provider: github
    url: https://api.github.com
    ...
  - provider: gitlab
     url: gitlab.com
    ...
``` 
which hit either of these cases.